elytron/WFCORE-4485-multiple-security-realms-distributed-identities.adoc

```diff
@@ -0,0 +1,66 @@
+= Support for multiple security realms - Distributed Identities
+:author:            Martin Mazanek
+:email:             mmazanek@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+
+== Overview
+
+By stacking LoginModules it was possible using PicketBox to attempt to authenticate using one remote store and if that failed try the next store in the list.
+
+This RFE will implement distributed security realm, which will allow to use multiple security realms for authentication by sequentially invoking list of configured realms until a realm with existing realm identity is found.
+
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.jboss.org/browse/WFCORE-4485[WFCORE-4485]
+
+=== Related Issues
+
+* https://issues.jboss.org/browse/ELY-1932[ELY-1932]
+
+=== Dev Contacts
+
+* mailto:mmazanek@redhat.com[Martin Mazanek]
+
+=== QE Contacts
+
+* TBD
+
+=== Testing By
+// Put an x in the relevant field to indicate if testing will be done by Engineering or QE.
+// Discuss with QE during the Kickoff state to decide this
+[ ] Engineering
+
+[ ] QE
+
+=== Affected Projects or Components
+
+* Wildfly Core
+* Elytron
+
+== Requirements
+
+=== Hard Requirements
+
+Add DistributedRealm to Elytron, which will be a SecurityRealm wrapper containing a list of SecurityRealms allowing to sequentially invoke them until one succeeds.
```

Comment:
What happens in case a realm is not available? Will the authentication fail in such case?